operators/pkg/controller/elasticsearch/driver/default.go

```diff
@@ -173,6 +174,8 @@ func (d *defaultDriver) Reconcile(
 		min = &d.Version
 	}
 
+	warnUnsupportedDistro(es, resourcesState.AllPods, d.Recorder)
```

Comment:
I think we have an event recorder already available in `reconcileState.Recorder`. It's a slightly different interface, but should work fine here.